router/internal/wsproto/subscriptions_transport_ws.go

```diff
@@ -119,6 +121,18 @@ func (p *subscriptionsTransportWSProtocol) WriteGraphQLErrors(id string, errors
 	})
 }
 
+func (p *subscriptionsTransportWSProtocol) Close(id string) error {
+	if err := p.conn.WriteCloser(ws.StatusGoingAway, "Downstream service error"); err != nil {
```

Comment:
When we call Close here, why with an error?

Reply:
It's a bit tricky semantically, I can add a comment to clarify

There are potentially other cases where we might want to call close with different codes eventually, but the only scenario this is used in right now is when the downstream subgraph closes without a complete message, which we _always_ transmute into a `1001: Downstream service error`